docs/source/scancode-toolkit/cli-reference/core-options.rst

```diff
@@ -0,0 +1,282 @@
+`Core` Options
+==============
+
+.. include::  /scancode-toolkit/rst_snippets/basic_options.rst
+
+.. _cli_core:
+
+.. include::  /scancode-toolkit/rst_snippets/core_options.rst
+
+``--generated`` Options
+-----------------------
+
+The ``--generated`` option classifies automatically generated code files with a flag.
+
+An example of using ``--generated`` in a scan::
+
+    ./scancode -clpieu --json-pp sample_generated.json samples --generated
+
+In the results, for each file the following attribute is added with it's corresponding True/False
+value ::
+
+    "is_generated": true
+
+For the samples folder, the following files have a true value for their is_generated attribute::
```

Comment:
Do you think it is a good idea to keep samples in the tool itself? I was willing to take the sample files out entirely and have a separate download for this.

Reply:
It's is a good question. I don't think it's done normally that this kind of sample folders are included with the software itself. Separating the sample folder and having a separate download will definitely be a better practice I think. The sample folder size is about ~1.2 MB, so it doesn't make a significant difference in the download size though. Even if you keep the folder inside scancode, having a separate download link will be great (perhaps even with multiple bigger samples) !

Reply:
I agree with separating out the samples - they are very old and not particularly interesting.

Reply:
Yes, If you could put together better and more interesting samples that'd be great.